### PR TITLE
[datafactory] `az datafactory factory create`: Enable managed identity by default

### LINKED ISSUE #2614: Identity is empty after creating Azure Datafactory with Azure DevOps pipeline
- If the issue is to do with Azure CLI 2.0 in-particular, create an issue here at [Azure/azure-cli](https://github.com/Azure/azure-cli/issues)

### Extension name (the extension in question)
datafactory                        0.2.0

### Description of issue (in as much detail as possible)
When creating a Azure datafactory using CLI and a service connection in Azure DevOps, the identity fields is $null

2020-11-03 16_21_51-Microsoft Store
2020-11-03 16_17_15-Pipelines - Run 20201103 2 logs

-----

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Resolve #2614 and align behavior with portal.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/datafactory/azext_datafactory/manual/custom.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if len(all_repo_configuration) > 1:
+        raise CLIError('At most one of --factory-vsts-configuration, --factory-git-hub-configuration is needed for '
+                       'repo configuration!')
+    repo_configuration = all_repo_configuration[0] if len(all_repo_configuration) == 1 else None
+    return client.create_or_update(resource_group_name=resource_group_name,
+                                   factory_name=factory_name,
+                                   if_match=if_match,
+                                   location=location,
+                                   tags=tags,
+                                   identity=FactoryIdentity(),
```
Comment (fengzhou-msft) [inline src/datafactory/azext_datafactory/manual/custom.py:38]: Changed from `identity=None`, `FactoryIdentity` has a `type` with a constant value `SystemAssigned`.

File: src/datafactory/azext_datafactory/manual/custom.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                               global_parameters=None):
+    from azext_datafactory.vendored_sdks.datafactory.models import FactoryIdentity
+    all_repo_configuration = []
+    if factory_vsts_configuration is not None:
+        all_repo_configuration.append(factory_vsts_configuration)
+    if factory_git_hub_configuration is not None:
+        all_repo_configuration.append(factory_git_hub_configuration)
+    if len(all_repo_configuration) > 1:
+        raise CLIError('At most one of --factory-vsts-configuration, --factory-git-hub-configuration is needed for '
+                       'repo configuration!')
```
Comment (fengzhou-msft) [inline src/datafactory/azext_datafactory/manual/custom.py:31]: Modified the generated message: `at most one of  factory_vsts_configuration, factory_git_hub_configuration is needed for repo_configuration!`

**Review by qiaozha — APPROVED** [review]

File: src/datafactory/azext_datafactory/manual/custom.py (hunk 1 above)
Comment (qiaozha) [reply src/datafactory/azext_datafactory/manual/custom.py:38]: I think in their newer swagger, they have extended identity type value

Comment (fengzhou-msft) [reply src/datafactory/azext_datafactory/manual/custom.py:38]: OK. We can add options to support that upon feature request.

**Pull request merged**